libfreerdp/core/transport.c

```diff
@@ -839,39 +884,61 @@ static int transport_default_read_pdu(rdpTransport* transport, wStream* s)
 	WINPR_ASSERT(transport);
 	WINPR_ASSERT(s);
 
-	/* Read in pdu length */
-	status = transport_parse_pdu(transport, s, &incomplete);
-	while ((status == 0) && incomplete)
+	/* RDS AAD Auth PDUs have no length indicator. We need to determine the end of the PDU by
```

Comment:
General comment: I think all the stuff around `transport_parse_pdu` needs some rework as I stated in the RDSTLS PR (#8758) about the double treatment (determining the size that already does some parsing, and then doing the effective parsing and partially redo the same job)